src/covid_model_seiir_pipeline/pipeline/forecasting/model/containers.py

```diff
@@ -74,6 +74,54 @@ def _build_index(start: pd.Series,
         return index
 
 
+@dataclass
+class ModelParameters:
+    # Core parameters
+    alpha: pd.Series
+    beta: pd.Series
+    sigma: pd.Series
+    gamma1: pd.Series
+    gamma2: pd.Series
+
+    # Theta parameters
+    theta_plus: pd.Series
+    theta_minus: pd.Series
+
+    # Vaccine parameters
+    unprotected_lr: pd.Series
+    protected_wild_type_lr: pd.Series
+    protected_all_types_lr: pd.Series
+    immune_wild_type_lr: pd.Series
+    immune_all_types_lr: pd.Series
+    old_unprotected_lr: pd.Series
+    old_protected_lr: pd.Series
+    old_immune_lr: pd.Series
+
+    unprotected_hr: pd.Series
+    protected_wild_type_hr: pd.Series
+    protected_all_types_hr: pd.Series
+    immune_wild_type_hr: pd.Series
+    immune_all_types_hr: pd.Series
+    old_unprotected_hr: pd.Series
+    old_protected_hr: pd.Series
+    old_immune_hr: pd.Series
+
+    # Variant parameters
+    beta_b117: pd.Series
+    beta_b1351: pd.Series
+    b117_prevalence: pd.Series
+    b1351_prevalence: pd.Series
+    probability_cross_immune: pd.Series
+
+    def with_index(self, index: pd.MultiIndex):
```

Comment:
This is more like `at_index` or `subset_to_index` than it is `with_index`

Perhaps to pedantic, but it seems like now would be the time...